pkg/api/v1alpha1/users.go

```diff
@@ -322,7 +362,7 @@ func (r *Router) updateUser(c *gin.Context) {
 		return
 	}
 
-	if req == (UserReq{}) {
+	if req.IsEmpty() {
```

Comment:
I hadn't noticed this before - I'm not a huge fan of adding more reflection although its pretty safe here. Maybe we're better doing the checks below and marking an `updated` bool that we check before starting the transaction?

Reply:
`go` won't let me use `==` to compare `UserNotificationPreferences`, since it's an array

Reply:
Sure, but why do we need this check at all?